packages/fetchai/connections/http_client/connection.py

```diff
@@ -263,16 +326,16 @@ def to_envelope(
         """
         context = EnvelopeContext(connection_id=connection_id)
         http_message = HttpMessage(
-            dialogue_reference=http_request_message.dialogue_reference,
-            target=http_request_message.target,
-            message_id=http_request_message.message_id,
```

Comment:
No; this should stay! The constructor is the right place to set these. The "update" approach should only be used for the newly added functionality (is_incoming, copy message, counterparty).

Reply:
done